### ISSUE TITLE
Favicon Icon upload form in Magento does not actually support .ico file types

### ISSUE THREAD

**Issue by ksenia-zlotin** [magento/magento2#34858]
When attempting to upload a favicon file of .ico extension type in Magento admin, Magento will not deem the file as valid.
Tested on a clean composer install of Magento 2.4.3-p1 with a PHP version of 7.4.13.

Perhaps the usage of PHP's `imagecreatefromstring` causes for an .ico file type to always be deemed invalid by Magento. The exception is thrown: "Warning: imagecreatefromstring(): one parameter to a memory allocation multiplication is negative or zero, failing operation gracefully
 in /var/www/html/vendor/magento/module-media-storage/Model/File/Validator/Image.php on line 64"

### Preconditions (*)
1. Unaltered, clean install of Magento version 2.4.3-p1
2. PHP version 7.4.13
3. DBMS: MySQL (ver. 5.5.5-10.2.40-MariaDB-1:10.2.40+maria~bionic)
Case sensitivity: plain=exact, delimited=exact
Driver: MySQL Connector/J (ver. mysql-connector-java-8.0.25 (Revision: 08be9e9b4cba6aa115f9b27b215887af40b159e0), JDBC4.2)
Effective version: MariaDB (ver. 10.2.40)
4. Chrome browser version Version 96.0.4664.110 (Official Build) (x86_64)
5. MacBook Pro - OS: Catalina 10.15.7

### Steps to reproduce (*)
1. Login to Magento admin
2. Navigate to Content -> Design -> Configuration
3. Click 'Edit' on any of the records shown in the grid
4. Scroll to 'Other Settings'
5. Expand 'HTML Head'
6. Under the 'Favicon Icon' section, click 'Upload'
7. Select a valid .ico file, click 'Open'

### Expected result (*)
1. It should provide a preview of the icon below the upload form and allow for the admin user to save the configuration settings.

### Actual result (*)
1. An error was shown to the admin:
![image](https://user-images.githubusercontent.com/62074773/146978236-23357f70-57f3-4b00-8a00-3d6264154dfe.png)

**Comment by engcom-Hotel** [comment]
Hi @ksenia-zlotin,
Thank you for reporting the issue.
We have tried this issue on Magento 2.4-develop and we have successfully reproduced it. Magento is not accepting the “.ico” extension files for the favicon icon event the ".ico" extension is in the allowed extension list.
Steps followed to reproduce issue:

1. log in to Magento admin
2. Navigate to Content -> Design -> Configuration
3. Click 'Edit' on any of the records shown in the grid
4. Scroll to 'Other Settings'
5. Expand 'HTML Head'
6. Under the 'Favicon Icon' section, click 'Upload'
7. Select a valid “.ico” extension file, click 'Open'
8. Observe the upload result (see the attached screenshots).
![image](https://user-images.githubusercontent.com/51681618/147225322-0ae5942b-1a4e-469d-a157-16911a25e70a.png)
![image](https://user-images.githubusercontent.com/51681618/147225445-a44146b9-f4d2-4073-bc6e-faa350d22af7.png)


Hence confirming the issue.



Thanks.
